## Apps Popularity: A study of iOS's and Android's app store

Our goal for this project is to analyze data to help developers understand what type of apps are likely to attract more users. The company only build apps that are free to download and install so their main source of revenue consists of in-app ads. This means that the revenue for any given app is mostly influenced by the number of users who use our app — the more users that see and engage with the ads, the better. 



 ### Reading and exploring the data

Rather than working with the whole data set of apps (more than 4000000!) we will work with a sample of all apps as provided by Dataquest: [This](https://dq-content.s3.amazonaws.com/350/googleplaystore.csv)  from the google play store and [this](https://dq-content.s3.amazonaws.com/350/AppleStore.csv) from the Apple app store 

In [8]:
opened_iOS = open('AppleStore.csv')
opened_Android = open('googleplaystore.csv')
from csv import reader
read_iOS = reader(opened_iOS)
read_Android = reader(opened_Android)
ios_data = list(read_iOS)
android_data = list(read_Android)


In order to explore the data, we will use this nice little function:

In [9]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

In [10]:
explore_data(ios_data,1,10)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


['429047995', 'Pinterest', '74778624', 'USD', '0.0', '1061624', '1814', '4.5', '4.0', '6.26', '12+', 'Social Networking', '37', '5', '27', '1']


['282935706', 'Bible', '92774400', 'USD', '0.0', '985920', '5320', '4.5', '5.0', '7.5.1', '4+', 'Reference', '37', '5', '45', '1']


['5538347

In [11]:
explore_data(ios_data,0,1,True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


Number of rows: 7198
Number of columns: 16


So there are 7198 apps in the sample and each one of them has 16 different [attributes](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps). Let's take a look at the apps from the google play store now:

In [12]:
explore_data(android_data,0,1,True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


Number of rows: 10842
Number of columns: 13


There are 10842 apps with 13 [attributes](https://www.kaggle.com/lava18/google-play-store-apps) each. Common attributes may have different names and compared with the ones before, there are some attribute missing. Let's see some of the forst lines of the file:

In [6]:
https://www.kaggle.com/lava18/google-play-store-appsexplore_data(ios_data,1,10)

SyntaxError: invalid syntax (<ipython-input-6-c6089feaf99e>, line 1)

It seems that the variables that would help the most in our analysis are: `Rating` and `Installs` (for the google play apps) and `user_ratings` and `rating_count_tot` on the apple side.

### Data Cleaning

#### Checking for incomplete information

There seems to be a problem with row 10472 of the google paystore date set. Let's verify that this is the case:

In [13]:
explore_data(android_data,10471,10474)

['Jazz Wi-Fi', 'COMMUNICATION', '3.4', '49', '4.0M', '10,000+', 'Free', '0', 'Everyone', 'Communication', 'February 10, 2017', '0.1', '2.3 and up']


['Xposed Wi-Fi-Pwd', 'PERSONALIZATION', '3.5', '1042', '404k', '100,000+', 'Free', '0', 'Everyone', 'Personalization', 'August 5, 2014', '3.0.0', '4.0.3 and up']


['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']




In fact is the row number 10473:

In [14]:
android_data[10473]

['Life Made WI-Fi Touchscreen Photo Frame',
 '1.9',
 '19',
 '3.0M',
 '1,000+',
 'Free',
 '0',
 'Everyone',
 '',
 'February 11, 2018',
 '1.0.19',
 '4.0 and up']

Better delete this now:

In [15]:
del android_data[10473]

In [16]:
android_data[10471:10474]

[['Jazz Wi-Fi',
  'COMMUNICATION',
  '3.4',
  '49',
  '4.0M',
  '10,000+',
  'Free',
  '0',
  'Everyone',
  'Communication',
  'February 10, 2017',
  '0.1',
  '2.3 and up'],
 ['Xposed Wi-Fi-Pwd',
  'PERSONALIZATION',
  '3.5',
  '1042',
  '404k',
  '100,000+',
  'Free',
  '0',
  'Everyone',
  'Personalization',
  'August 5, 2014',
  '3.0.0',
  '4.0.3 and up'],
 ['osmino Wi-Fi: free WiFi',
  'TOOLS',
  '4.2',
  '134203',
  '4.1M',
  '10,000,000+',
  'Free',
  '0',
  'Everyone',
  'Tools',
  'August 7, 2018',
  '6.06.14',
  '4.4 and up']]

It really deleted the row. 
Now, let's examine the iOs data to see if there is any problem. No problem there.


#### Checking for duplicate entries

There seems to be duplicated rows in the google apps data set. Let's see if it is true

In [17]:
duplicated_apps = []
unique_apps = []
for apps in android_data:
    name = apps[0]
    if name in unique_apps:
        duplicated_apps.append(name)
    else:
        unique_apps.append(name)
print('Number of duplicated apps: ',len(duplicated_apps))
print("\n")
print('Example of duplicated apps:', duplicated_apps[:15])
print("\n")
print('Unique apps: ',len(unique_apps))


Number of duplicated apps:  1181


Example of duplicated apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


Unique apps:  9660


Let's see as an example some rows with duplicated entries:

In [18]:
for apps in android_data:
    name = apps[0]
    if name == 'Box':
        print(apps)


['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']


We need to eliminate redundant rows.The criteria used will be the one that has the most installs. However, this criteria does not seem to help in the previous case. It does seem to help, for example in the case of Instagram:

In [19]:
for apps in android_data:
    name = apps[0]
    if name == 'Instagram':
        print(apps)


['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


In this case, we will keep the last one.

#### Removing the extra lines

In [20]:
reviews_max = {}
for apps in android_data[1:]:
    name = apps[0]
    n_reviews = float(apps[3])
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews

Let's check to see if we indeed remove duplicated apps:

In [21]:
len(reviews_max)

9659

Good!. That is what we should get. Now the second part:

In [22]:
android_clean =[]
already_added = []
for apps in android_data[1:]:
    name = apps[0]
    n_reviews = float(apps[3])
    if n_reviews == reviews_max[name] and name not in already_added:
        android_clean.append(apps)
        already_added.append(name)

Letś check to see if everything is ok:

In [24]:
len(already_added)    
explore_data(android_clean,1,4)
explore_data(android_clean,9654,9659)

['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Sya9a Maroc - FR', 'FAMILY', '4.5', '38', '53M', '5,000+', 'Free', '0', 'Everyone', 'Education', 'July 25, 2017', '1.48', '4.1 and up']


['Fr. Mike Schmitz Audio Teachings', 'FAMILY', '5.0', '4', '3.6M', '100+', 'Free', '0', 'Everyone', 'Education', 'July 6, 2018', '1.0', '4.1 and up']


['Parkinson Exercices FR', 'MEDICAL', 'NaN', '3', '9.5M', '1,000+', 'Free', '0', 'Everyone', 'Medical', 'January 20, 2017', '1.0', '2.2 and up']


['Th

Looks good!. Now into the next step

### Removing non-english apps


This function detects if a given word is an english word. It assumes that english characters have always ASCII values less than o equal to 127

In [25]:
def detect_english(string):
    english = True
    for caracter in string:
        if ord(caracter) > 127:
            english = False
    return english

In [26]:
print(detect_english('maría'))
print(detect_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(detect_english('Instagram'))
print(detect_english('Instachat 😜'))
print(detect_english('Docs To Go™ Free Office Suite'))

False
False
True
False
False


On the previous screen, we wrote a function that detects non-English app names, but we saw that the function couldn't correctly identify certain English app names like 'Docs To Go™ Free Office Suite' and 'Instachat 😜'. This is because emojis and characters like ™ fall outside the ASCII range and have corresponding numbers over 127.

If we're going to use the function we've created, we'll lose useful data since many English apps will be incorrectly labeled as non-English. To minimize the impact of data loss, we'll only remove an app if its name has more than three characters with corresponding numbers falling outside the ASCII range. This means all English apps with up to three emoji or other special characters will still be labeled as English. Our filter function is still not perfect, but it should be fairly effective.

Let's edit the function we created in the previous screen, and then use it to filter out the non english name apps

In [27]:
def detect_english(string):
    english = True
    n = 0
    for caracter in string:
        if ord(caracter) > 127:
            n += 1
            if n > 3:
                english = False
    return english

print(detect_english('Docs To Go™ Free Office Suite'))
print(detect_english('Instachat 😜'))
print(detect_english('María'))


True
True
True


Now, we will use this function to filter out the list of both apps to only keep the ones with english names. Starting with the android apps:

In [28]:
android_eng = []
for app in android_clean[0:]:
    name = app[0]
    if detect_english(name):
        android_eng.append(app)
print(len(android_eng))
explore_data(android_eng,0,9)

9614
['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


['Smoke Effect Photo Maker - Smoke Editor', 'ART_AND_DESIGN', '3.8', '178', '19M', '50

Now, I will do the same with the Apple store apps. Note that the name is located in app[1] and not in app[0], as before. Notice also that we are left with 9614 android apps and 6183 apple apps

In [29]:
apple_eng = []
for app in ios_data[1:]:
    name = app[1]
    if detect_english(name):
        apple_eng.append(app)
print(len(apple_eng))
print(len(ios_data[1:]) - len(apple_eng))
explore_data(apple_eng,0,9)

6183
1014
['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


['429047995', 'Pinterest', '74778624', 'USD', '0.0', '1061624', '1814', '4.5', '4.0', '6.26', '12+', 'Social Networking', '37', '5', '27', '1']


['282935706', 'Bible', '92774400', 'USD', '0.0', '985920', '5320', '4.5', '5.0', '7.5.1', '4+', 'Reference', '37', '5', '45', '1']



### Isolate free apps

First, with the ios apps:

In [30]:
apple_free = []
for app in apple_eng:
    price = app[4]
    #print(price)
    if price == '0.0':
        apple_free.append(app)
print(len(apple_free))
print(len(apple_eng))
explore_data(apple_free,0,9)

3222
6183
['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


['429047995', 'Pinterest', '74778624', 'USD', '0.0', '1061624', '1814', '4.5', '4.0', '6.26', '12+', 'Social Networking', '37', '5', '27', '1']


['282935706', 'Bible', '92774400', 'USD', '0.0', '985920', '5320', '4.5', '5.0', '7.5.1', '4+', 'Reference', '37', '5', '45', '1']



Now, with the google play store apps:

In [49]:
android_free = []
for app in android_eng:
    price = app[7]
    #print(price)
    if price == '0':
        android_free.append(app)
print(len(android_free))
print(len(android_eng))
explore_data(android_free,0,9)

8864
9614
['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


['Smoke Effect Photo Maker - Smoke Editor', 'ART_AND_DESIGN', '3.8', '178', '19M'

### Most common apps by genre: Part 1 (step 9 of 14!)

As we mentioned in the introduction, our aim is to determine the kinds of apps that are likely to attract more users because our revenue is highly influenced by the number of people using our apps.

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:

1. Build a minimal Android version of the app, and add it to Google Play.
2. If the app has a good response from users, we develop it further.
3. If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.

Because our end goal is to add the app on both Google Play and the App Store, we need to find app profiles that are successful on both markets. For instance, a profile that works well for both markets might be a productivity app that makes use of gamification.

Let's begin the analysis by getting a sense of what are the most common genres for each market. For this, we'll need to build frequency tables for a few columns in our data sets.

For the iOS apps, the field related to genre has the index 12 and for the android apps has the index 1.

But first, we need to calculate those frequency tables.

In [54]:
def freq_table(dataset,index):
    freq_dic={}
    perc_dic={}
    total = 0
    for row in dataset:
        category = row[index]
        total +=1
        if category in freq_dic:
            freq_dic[category] += 1
        else:
            freq_dic[category] = 1
    for key in freq_dic:
        percentage = (freq_dic[key]/total)*100
        perc_dic[key] = percentage
        #print(percentage)
    
    return perc_dic

#freq_table(android_free,1)


In [51]:
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])
        


Let see the results for Android category field:

In [55]:
display_table(android_free,1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

Family is the winning category, followed closely by game and tools. There is also a "genre"category. The results for this one are:

In [56]:
display_table(android_free,9)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

Tools and entertainment are the most popular genre apps. 

In the case of the Apple store, this is what we get for the variable `prime genre`

In [57]:
display_table(apple_free,11)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


In the case of the Apple store, games and entertainment seem to be the most popular categories. It sees that Android users have more diverse interest than apple users.

### Step 12 of 14: Most popular apps by genre on App store

### Step 13 of 14: Most popular apps by genre on Android